src/main/java/seedu/address/model/entity/shop/Shop.java

```diff
@@ -150,28 +149,29 @@ public PartMap getPartMap() {
     }
 
     /**
-     * Adds part to the part map
+     * Adds part
      *
-     * @param part Part to be added
+     * @param partName Name of the part to add
+     * @param quantity Quantity of the part to add
      */
-    public void addPart(Part part) {
-        this.getPartMap().addPart(part.getName(), part);
+    public void addPart(String partName, int quantity) {
+        this.getPartMap().addPart(partName, quantity);
     }
 
     /**
-     * Increases part stock
+     * Increases part stock by a specified quantity
      *
      * @param partName Name of part
-     * @param amt      Amount to increase by
+     * @param quantity Quanity to increase by
      */
-    public void addPartStock(String partName, int amt) throws PartNotFoundException {
-        this.partMap.getPart(partName).increaseStock(amt);
+    public void addPartStock(String partName, int quantity) throws PartNotFoundException {
+        this.partMap.increasePartQuantity(partName, quantity);
```

Comment:
increasePartQuantity does not throw PartNotFoundException. I don't think need to throw now and can just use addPart()